server/tests/ballot_comparison/test_ballot_comparison.py

```diff
@@ -161,6 +161,13 @@ def test_ballot_comparison_two_rounds(
     sample_size = sample_size_options[target_contest_id][0]
     snapshot.assert_match(sample_size)
 
+    contest = Contest.query.get(target_contest_id)
+    assert contest.total_ballots_cast is None
+    assert contest.votes_allowed is None
+    assert contest.choices == []
+
+    set_contest_metadata_from_cvrs(contest)
+
```

Comment:
why did you add this block?

Reply:
ah, i think i understand. now that we're validating the sample size against the contest metadata, it needs to be set before running that validation. this means there's actually an implicit assumption in the create round endpoint that the contest metadata has already been set. i think we should make that explicit so there's no hidden failure case.

My recommendation would be to move the block of code that sets the metadata up from later in the create rounds endpoint to before the validation. This is the block:

```python
    # For ballot comparison audits, we need to lock in the contest metadata we
    # parse from the CVRs when we launch the audit.
    if (
        election.audit_type == AuditType.BALLOT_COMPARISON
        and json_round["roundNum"] == 1
    ):
        for contest in election.contests:
            set_contest_metadata_from_cvrs(contest)
```

Will probably have to be tweaked to `json_round.get("roundNum")` since it will come before we actually validate the JSON. A bit weird, but I think it should work fine.

Another option would be to split the validation into two functions - one for the round generally, and one for the sample sizes, and then call the sample size validation function after setting the contest metadata.

Reply:
@jonahkagan Okay, I went for the second option, let me know what you think!

Reply:
can you remove this block from all the tests now?